torch/fx/symbolic_trace.py

```diff
@@ -187,6 +196,9 @@ def _proxy_placeholder(name: str, delegate: DelegateBase) -> Proxy:
 #   - root - the `nn.Module` instance to trace
 #   - delegate : An instance of a Delegate object
 def symbolic_trace(root : torch.nn.Module, delegate_class=DefaultDelegate) -> GraphModule:
+
+    global _FX_IS_TRACING
+    _FX_IS_TRACING = True
```

Comment:
I'd enforce that we currently not in tracing mode already, just in case :)

Reply:
updated. required another nesting of `try... finally`. let me know if you know of a better way 😅

Reply:
I think you can keep a check on whether we're tracing already here but then move setting way down next to the existing `try:` and undo it in existing `finally:`. The only place where we run the user code is `graph.output(delegate.create_arg(fn(*args)))` so we need to wrap only that

Reply:
✅